### PR TITLE
Issue #31: amp: bucket replication

### PR DESCRIPTION
**Pull request by danielcahoon**

### CODE REVIEW

File: pkg/amp/bucket_sets.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	"github.com/cisco/arc/pkg/log"
+	"github.com/cisco/arc/pkg/msg"
+	"github.com/cisco/arc/pkg/provider"
+	"github.com/cisco/arc/pkg/resource"
+	"github.com/cisco/arc/pkg/route"
+)
+
+type bucketSets struct {
+	*resource.Resources
+	BucketSets []*config.BucketSet
```
Comment (icahoon) [inline pkg/amp/bucket_sets.go]: Is this required? It looks like the config for the bucket sets is only used in the constructor.

File: pkg/amp/bucket_sets.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	bucketSets map[string]resource.BucketSet
+	storage    *storage
+}
+
+func newBucketSets(s *storage, prov provider.Account, cfg []*config.BucketSet) (*bucketSets, error) {
+	log.Debug("Initializing Bucket Sets")
+
+	b := &bucketSets{
+		Resources:  resource.NewResources(),
+		BucketSets: cfg,
```
Comment (icahoon) [inline pkg/amp/bucket_sets.go]: This can probably be removed

**Review by kwelch426 — APPROVED** [review]

File: pkg/aws/bucket.go [hunk 3]
```diff
@@ -124,56 +124,32 @@ func (b *bucket) enableEncryption() error {
...
+		return fmt.Errorf("No Role or Destination found for replication")
+	}
+	params := &s3.PutBucketReplicationInput{
+		Bucket: aws.String(b.Name()),
+		ReplicationConfiguration: &s3.ReplicationConfiguration{
+			Role: aws.String(b.arn.String()),
+			Rules: []*s3.ReplicationRule{
+				{
+					Destination: &s3.Destination{
+						Bucket: aws.String("arn:aws:s3:::" + b.Destination()),
```
Comment (icahoon) [inline pkg/aws/bucket.go]: should you be using the arn code here rather than hardcoding it?

File: pkg/aws/bucket.go [hunk 4]
```diff
@@ -124,56 +124,32 @@ func (b *bucket) enableEncryption() error {
...
 					},
+					Prefix: aws.String(""),
```
Comment (icahoon) [inline pkg/aws/bucket.go]: What does Prefix default to? Perhaps this isn't needed?

**Review by icahoon — CHANGES_REQUESTED** [review]
This is looking great. Just some minor comments.

**Review by icahoon — APPROVED** [review]
lgtm. thanks for making the changes.

**Comment by icahoon** [comment]
this closes issue #31

**Pull request merged**